include/irr/core/IReferenceCounted.h

```diff
@@ -384,5 +384,18 @@ namespace core
 }
 } // end namespace irr
 
+namespace std {
+
+    template <typename T>
+    struct hash<irr::core::smart_refctd_ptr<T>>
+    {
+        std::size_t operator()(const irr::core::smart_refctd_ptr<T>& k) const
+        {
+            return reinterpret_cast<std::size_t>(k.get());
+        }
+    };
+
+}
+

```

Comment:
adding things to `std::` namespace is UB, we should just provide a hash functor that can be used as a template parameter

Reply:
thought the same but seems like it's common practice in case of has<>, not sure if legal. btw there's a lot of this in irrbaw :p

Reply:
recently removed some of it.

Reply:
https://en.cppreference.com/w/cpp/language/extending_std
"Adding template specializations" section

Reply:
also sample code in std::hash docs encourage this https://en.cppreference.com/w/cpp/utility/hash

Reply:
all right, I give up